pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -729,13 +748,23 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(podInfoList []*framework.
 			gated := pInfo.Gated
 			if added, _ := p.addToActiveQ(pInfo); added {
 				klog.V(5).InfoS("Pod moved to an internal scheduling queue", "pod", klog.KObj(pInfo.Pod), "event", event, "queue", activeQName)
+				pInfo.MoveRequestCycle = p.schedulingCycle
 				activated = true
 				metrics.SchedulerQueueIncomingPods.WithLabelValues("active", event.Label).Inc()
 				p.unschedulablePods.delete(pod, gated)
 			}
 		}
 	}
-	p.moveRequestCycle = p.schedulingCycle
+
+	// Besides setting MoveRequestCycle above for all pods which got moved,
+	// we also need to set it for pods that are being processed because
+	// AddUnschedulableIfNotPresent might get called for them, and in
+	// AddUnschedulableIfNotPresent we need to know whether events where
+	// observed while processing them.
+	for uid := range p.inFlightPods {
+		p.inFlightPods[uid] = p.schedulingCycle
+	}
```

Comment:
Can this be moved inside `if activated`?

Reply:
No. If there's a single pod in flight and some cluster event comes in that affects the pod, that pod must go into the backoff queue if pod scheduling fails. But `activated` would be false in that case because there was no other pod that got activated.

Reply:
While squashing, I was reminded that *not* forcing in-flight pods into the backoff queue unnecessarily was originally the goal for this PR. As seen here, the PR no longer achieves that goal when setting `p.inFlightPods[uid]` for all in-flight pods.

But I [was seeing a benefit](https://github.com/kubernetes/kubernetes/pull/115796#issuecomment-1442134651) from this PR in my integration test branch for DRA. I'm now puzzled why exactly that is, so I double-checked my benchmarking. I think what had happened was that I botched a rebase and ran the benchmark while it didn't set the move request cycle for pods in-flight - that gives good results, but isn't correct :cry: 

The good news is that this PR here can be fixed, it just needs another improvement first that I also have in my full branch: letting plugins register a filter function together with the cluster events that they are interested in. Then in this code here, an unrelated cluster event will not affect an in-flight pod.

I suggest that we change the order of PRs:
- first some general scheduler_perf enhancements (new PR)
- adding DRA testcases to it without any optimization (beware, this will increase overall runtime!)
- cluster event filter callbacks (new PR)
- this PR

Then in each PR which aims at enhancing performance, the impact can be measured with "benchstat" before merging.

/hold

Reply:
> letting plugins register a filter function together with the cluster events that they are interested in.

That sounds like a non-trivial change :) but as long as each PR is well organized and we can tell the perf diff, we should be confident enough to evaluate it.